lib/r10k/action/deploy/module.rb

```diff
@@ -69,6 +70,19 @@ def call
 
         def visit_deployment(deployment)
           yield
+        ensure
+          if (postcmd = @settings[:postrun]) && @modified_envs.any?
+            if postcmd.grep('$modifiedenvs').any?
+              envs_to_run = @modified_envs.join(' ')
+              logger.debug "Running postrun command for environments #{envs_to_run}."
+              postcmd = postcmd.map { |e| e.gsub('$modifiedenvs', envs_to_run) }
+            end
+            subproc = R10K::Util::Subprocess.new(postcmd)
+            subproc.logger = logger
+            subproc.execute
+          else
+            logger.debug "No environments were modified, not executing postrun command."
```

Comment:
This could happen because environments were modified but the postrun setting is nil.

Reply:
Oh good catch, I'll split that out.

Reply:
Updated.